/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.checkerframework.checker.nullness.qual.Nullable
 */
package com.viaversion.viaversion.api.protocol.packet.mapping;

import com.viaversion.viaversion.api.protocol.packet.PacketType;
import com.viaversion.viaversion.api.protocol.packet.State;
import com.viaversion.viaversion.api.protocol.packet.mapping.PacketArrayMappings;
import com.viaversion.viaversion.api.protocol.packet.mapping.PacketMapping;
import org.checkerframework.checker.nullness.qual.Nullable;

public interface PacketMappings {
    public @Nullable PacketMapping mappedPacket(State var1, int var2);

    public static PacketMappings arrayMappings() {
        return new PacketArrayMappings();
    }

    default public void addMapping(PacketType packetType, PacketMapping mapping) {
        this.addMapping(packetType.state(), packetType.getId(), mapping);
    }

    public void addMapping(State var1, int var2, PacketMapping var3);

    default public boolean hasMapping(PacketType packetType) {
        return this.mappedPacket(packetType.state(), packetType.getId()) != null;
    }

    default public boolean hasMapping(State state, int unmappedId) {
        return this.mappedPacket(state, unmappedId) != null;
    }
}

